### PR TITLE
Modularize SanitizedFile#copy_to and SanitizedFile#move_to

### PR DESCRIPTION
**Pull request by oss92**
I have edited some parts to decrease complexity and code duplication.

### CODE REVIEW

File: lib/carrierwave/processing/mini_magick.rb [hunk 1]
```diff
@@ -184,15 +184,10 @@ def resize_to_fill(width, height, gravity = 'Center', combine_options: {})
...
-              cmd.resize "#{cols}"
-            else
-              cols = (scale_y * (cols + 0.5)).round
-              rows = (scale_y * (rows + 0.5)).round
-              cmd.resize "x#{rows}"
-            end
+            reference_scale = scale_x >= scale_y ? scale_x : scale_y
+            cols = (reference_scale * (cols + 0.5)).round
+            rows = (reference_scale * (rows + 0.5)).round
+            reference_scale == scale_x ? cmd.resize("#{cols}") : cmd.resize("x#{rows}")
```
Comment (thomasfedb) [inline lib/carrierwave/processing/mini_magick.rb]: Is this assignment required?

**Comment by thomasfedb** [comment]
I like the changes to `SanitizedFile`. The others I might need a little more convincing of.

**Comment by oss92** [comment]
As for the `Mounter`, The `init_uploaders` method saved a lot of code duplication.

As for `MiniMagick`, I guess assignment would be required. Maybe change `reference_scale = scale_x >= scale_y ? scale_x : scale_y` to `scale_x = scale_y > scale_x ? scale_y : scale_x` if you do not want an extra variable. What do you think?

**Comment by thomasfedb** [comment]
Perhaps something more like this:

```ruby
scale_x, scale_y = width / cols.to_f, height / rows.to_f
scale = [scale_x, scale_y].max
cols, rows = [cols, rows].map {|dim| dim * scale }
```

**Comment by oss92** [comment]
I thought about this, but then you will have to deal with this `scale == scale_x ? cmd.resize("#{cols}") : cmd.resize("x#{rows}")`. 

Anyways changed it.

**Comment by thomasfedb** [comment]
@oss92 Would you be able to split out the changes to `SanitizedFile`? I'd be very happy to merge those in without much trouble. The others might take a bit longer.

**Comment by oss92** [comment]
Done. Updated this pull request and will create another for the other changes if you'd like.

**Pull request merged**